### PR TITLE
fix: LSDV-1023: Add audio error handler to audio v3

### PR DESCRIPTION
**Pull request by juliosgarbi**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_
Screenshot 2023-02-13 at 11 16 38 AM
Added audio error handler like it is for Audio V1

#### What does this fix?
If user try to load a file that is not decoded, an unsupported file extension, or the audio V3 got some error during the decoding and file loading, it shows an error handler like audio V1

#### What libraries were added/updated?
I had to change audio-file-decoder to @martel/audio-file-decoder because audio-file-decoder is stucking on promise when it's initializating, so @bmartel forked audio-file-decoder and fixed the problem that we were having.

#### Does this change affect performance?
no

#### Does this change affect security?
no

#### What alternative approaches were there?
na

#### What feature flags were used to cover this change?
na

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [ ] integration
- [ ] unit

### CODE REVIEW

File: src/utils/messages.js [hunk 1]
```diff
@@ -54,13 +54,11 @@ export default {
...
   ERR_LOADING_AUDIO({ attr, url, error }) {
     return (
-      <p>
-        Error while loading audio. Check <code>{attr}</code> field in task.
-        <br />
-        Technical description: {error}
-        <br />
-        URL: {url}
-      </p>
+      <div>
```
Comment (bmartel) [inline src/utils/messages.js:57]:
```suggestion
      <div data-testid="error:audio">
```

Maybe we make this explicit so we can update the checks for errors in test to be explicit. That way we know for sure this error is getting shown, not just the one in the Audio interface itself.

**Review by bmartel — CHANGES_REQUESTED** [review]

File: e2e/fragments/AtAudioView.js [hunk 2]
```diff
@@ -137,6 +137,10 @@ module.exports = {
     I.click(this._playButtonSelector);
   },
 
+  seeErrorHandler(value) {
+    I.see(value);
```
[the remark this answers is not in the document]
Comment (bmartel) [reply e2e/fragments/AtAudioView.js:141]: And also in here we would define `_errorSelector` to `[data-testid="error:audio"]`

File: src/tags/object/AudioUltra/view.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
 import { observer } from 'mobx-react';
-import { FC, useEffect, useRef } from 'react';
+import React, { FC, useEffect, useRef } from 'react';
```
Comment (bmartel) [inline src/tags/object/AudioUltra/view.tsx:2]:
```suggestion
import { FC, useEffect, useRef } from 'react';
```
Are we using this `React` import anywhere?

Comment (juliosgarbi) [reply src/tags/object/AudioUltra/view.tsx]: No, my IDE imported it, but I don't know why!

**Review by bmartel — APPROVED** [review]

File: src/tags/object/AudioUltra/model.js [hunk 4]
```diff
@@ -440,7 +440,9 @@ export const AudioModel = types.compose(
         },
 
         onError(error) {
-          self.errors = [error];
+          const message = getEnv(self.store).messages.ERR_LOADING_AUDIO({ attr: self.value, url: self._value, error:error.message });
```
Comment (bmartel) [inline src/tags/object/AudioUltra/model.js:443]:
```suggestion
          const message = getEnv(self.store).messages.ERR_LOADING_AUDIO({ attr: self.value, url: self._value, error: error.message });
```

**Comment by bmartel** [comment]
@juliosgarbi The e2e are failing here, I would like to see them pass before I ✅ the related repos.

**Review by hlomzik — APPROVED** [review]

File: e2e/tests/audio/audio-errors.test.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    annotations: [{ id: 'test', result: annotations }],
+    config,
+    data: {
+      url: '/files/doesnt_exist.mp3',
+    },
+  });
+
+  await AtAudioView.lookForStage();
+
+  AtAudioView.seeErrorHandler('HTTP error status: 404');
```
Comment (bmartel) [inline e2e/tests/audio/audio-errors.test.js:80]:
```suggestion
  AtAudioView.seeErrorHandler('HTTP error status: 404', this._httpErrorSelector);
```

**Review by bmartel — CHANGES_REQUESTED** [review]
The http error can be used for various things (Timeseries, Video, etc.), let's make the testid reflect the error type instead of where we are receiving it.

**Review by bmartel — APPROVED** [review]

**Pull request merged**